metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -358,3 +443,99 @@ def get_fk_constraints_for_schema(
             constraints[row["fk_table_name"]].append(fk_constraints_map[row["fk_name"]])
 
         return constraints
+
+    def get_tags_for_database_without_propagation(
+        self,
+        db_name: str,
+    ) -> _SnowflakeTagCache:
+        cur = self.query(
+            SnowflakeQuery.get_all_tags_in_database_without_propagation(db_name)
+        )
+
+        tags = _SnowflakeTagCache()
+
+        for tag in cur:
+            snowflake_tag = SnowflakeTag(
+                database=tag["TAG_DATABASE"],
+                schema=tag["TAG_SCHEMA"],
+                name=tag["TAG_NAME"],
+                value=tag["TAG_VALUE"],
+            )
+
+            # This is the name of the object, unless the object is a column, in which
+            # case the name is in the `COLUMN_NAME` field.
+            object_name = tag["OBJECT_NAME"]
+            # This will be null if the object is a database or schema
+            object_schema = tag["OBJECT_SCHEMA"]
+            # This will be null if the object is a database
+            object_database = tag["OBJECT_DATABASE"]
+
+            domain = tag["DOMAIN"]
```

Comment:
Can we extend `SnowflakeObjectDomain` from [snowflake/constants.py ](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/snowflake/constants.py#L34) to add database and schema  and use it here instead of raw strings ?